python/src/uagents/agent.py

```diff
@@ -294,12 +300,51 @@ def _initialize_wallet_and_identity(self, seed, name):
         else:
             self._identity = Identity.from_seed(seed, 0)
             self._wallet = LocalWallet(
-                PrivateKey(derive_key_from_seed(seed, LEDGER_PREFIX, 0)),
+                PrivateKey(
+                    derive_key_from_seed(
+                        seed, LEDGER_PREFIX, wallet_key_derivation_index
+                    )
+                ),
                 prefix=LEDGER_PREFIX,
             )
         if name is None:
             self._name = self.address[0:16]
 
+    def initialize_wallet_messaging(
+        self, enable_wallet_messaging: Union[bool, Dict[str, str]]
```

Comment:
What's the idea behind `Dict[str, str]` -> what values do we expect to come up here?
{"chain_id": "???"} 

Also, as with the `endpoint` and other definitions: the myriad ways to instantiate an agent are not well documented. We need to keep that in mind when providing such flexible interfaces.

Reply:
Yes, I believe this is primarily to allow configuring the chain id, as the messaging server needs to know this. But since only our own testnet and mainnet are allowed, I'm not sure it makes sense to expose this in the API.

Reply:
OK, I think this is because the chain id used for wallet messaging can be mainnet even while the agent is only transacting on the testnet, so we needed a way to configure this separately.

Reply:
Added this explanation to the docstring.

Reply:
I think it would be sensible to bind the messaging to the network the agent is transacting on to somewhat force developers when we switch to main net.
At least I don't see a reason for us to support agents running on one net and sending messages on the other.

Reply:
At the moment, you can set the chain in the Fetch wallet but our agents are only running on the Dorado testnet, so we needed them to be able to communicate on mainnet even though they register on testnet. I agree that we will need to revisit this when migrating to mainnet, though.